lighthouse-cli/test/smokehouse/smokehouse-report.js

```diff
@@ -103,46 +103,73 @@ function findDifference(path, actual, expected) {
 }
 
 /**
- * Collate results into comparisons of actual and expected scores on each audit.
- * @param {Smokehouse.ExpectedLHR} actual
- * @param {Smokehouse.ExpectedLHR} expected
- * @return {Smokehouse.LHRComparison}
+ * @param {string} name – name of the value being asserted on (e.g. the result of a certain audit)
+ * @param {any} actualResult
+ * @param {any} expectedResult
+ * @return {Smokehouse.Comparison}
+ */
+function makeComparison(name, actualResult, expectedResult) {
+  const diff = findDifference(name, actualResult, expectedResult);
+
+  return {
+    name,
+    actual: actualResult,
+    expected: expectedResult,
+    equal: !diff,
+    diff,
+  };
+}
+
+/**
+ * Collate results into comparisons of actual and expected scores on each audit/artifact.
+ * @param {Smokehouse.ExpectedRunnerResult} actual
+ * @param {Smokehouse.ExpectedRunnerResult} expected
+ * @return {Smokehouse.Comparison[]}
  */
 function collateResults(actual, expected) {
-  const auditNames = Object.keys(expected.audits);
-  const collatedAudits = auditNames.map(auditName => {
-    const actualResult = actual.audits[auditName];
+  /** @type {Smokehouse.Comparison[]} */
+  let artifactAssertions = [];
+  if (expected.artifacts) {
+    const artifactNames = /** @type {(keyof LH.Artifacts)[]} */ (Object.keys(expected.artifacts));
+    artifactAssertions = artifactNames.map(artifactName => {
+      const actualResult = (actual.artifacts || {})[artifactName];
+      if (!actualResult) {
+        throw new Error(`Config run did not generate artifact ${artifactName}`);
+      }
+
+      const expectedResult = (expected.artifacts || {})[artifactName];
```

Comment:
should be able to nuke this `||` given the if on line 132

Reply:
Hmm, TS fails without the `|| {}` if the artifacts are accessed in the `map` callback. In general constraining the type seems to work in nested functions, but not here somehow...

Reply:
ah right the TS callback problem, my preferred way around this recently has been to assign `expected.artifacts` to a variable right after we've asserted its definedness, then when you access it in the closure TS already knows it's safe